monzo.js

```diff
@@ -1,133 +1,239 @@
-const PD = require("probability-distributions");
-
-const GOAL = 250000;
-const PLEDGE_VALUES = [10,20,50,100,250,500,1000];
-const DATASET = PD.sample(PLEDGE_VALUES, 10000, true, [7,6,5,4,3,2,1]);
-// Convert to object so each pledge has a unique ID
-var allPledges = DATASET.reduce(function(obj, currentValue, index) {
-  obj[index] = currentValue;
-  return obj;
-}, {});
-
-var allPledgesTotal = DATASET.reduce(function (a, b) { return a + b; }, 0);
-
-console.log("Goal: " + GOAL +
-  "\nTotal pledged: " + allPledgesTotal
-);
-if (allPledgesTotal < GOAL) {
-  console.log("NOT ENOUGH PLEDGES TO REACH GOAL");
-  return;
-}
-
-var runningTotal = allPledgesTotal;
-
-var successfulApplicants = getCopyOfObject(allPledges);
-
-chooseSuccessfulApplicants();
-
-removeObjectSubset(successfulApplicants, allPledges);
-
-
+(function() {
+
+// Set global namespace object
+window.monzo = monzo = {};
+
+// To be set directly by passed arguments on execution, and treated as constants
+var goal,
+    pledgeValues,
+    testData;
+
+// Main variables
+var allPledges,
+    allPledgesTotal,
+    runningTotal,
+    successfulApplicants,
+    numberOfFailed;
+
+// For external programs to see if variables have already been loaded
+var areVariablesLoaded = false;
+
+
+/*---------------PUBLIC FUNCTIONS---------------*/
+
+monzo.getGoal = function () {
+  return Number(goal).toLocaleString("en")
```

Comment:
it would return "NaN", as would the other get methods. Is this a problem though? These methods need to be public, so not sure how i would prevent outside scripts running them before initilisation?